hil/ext/switches/ovs.py

```diff
@@ -0,0 +1,248 @@
+            )
+    hostname = db.Column(db.String, nullable=False)
+
+    @staticmethod
+    def validate(kwargs):
+        """Checks input to match switch parameters."""
+        schema.Schema({
+            'hostname': basestring,
+        }).validate(kwargs)
+
+# 1. Public Methods:
+
+    def get_capabilities(self):
+        return ['nativeless-trunk-mode']
+
+    def ovs_connect(self, *command_strings):
+        """Interacts with the Openvswitch.
+
+        Args:
+            *command_strings (tuple) : tuple of shell commands required
+                        to make changes to openvswitch
+        Raises: SwitchError
+        Returns: If successful returns None else logs error message
+        """
+        try:
+            for command in command_strings:
+                arg_list = shlex.split(command)
+                subprocess.check_call(arg_list)
+        except subprocess.CalledProcessError as e:
+            logger.error('%s', e)
+            raise SwitchError('ovs command failed: ')
+
+    def get_port_networks(self, ports):
+
+        response = {}
+        for port in ports:
+            trunk_id_list = self._interface_info(port.label)['trunks']
+            if trunk_id_list == []:
+                response[port] = []
+            else:
+                result = []
+                for trunk in trunk_id_list:
+                    name = "vlan/"+trunk
+                    result.append((name, trunk))
+                response[port] = result
+            native = self._interface_info(port.label)['tag']
+            if native != []:
+                response[port].append(("vlan/native", native))
+
+        return response
+
+    def revert_port(self, port):
+
+        shell_cmd_1 = 'sudo ovs-vsctl del-port {port}'.format(port=port)
+        shell_cmd_2 = 'sudo ovs-vsctl add-port {switch} {port}'.format(
+                    switch=self.hostname, port=port
+                    ) + ' vlan_mode=native-untagged'
+        self.ovs_connect(shell_cmd_1, shell_cmd_2)
+
+    def modify_port(self, port, channel, new_network):
+
+        (port,) = filter(lambda p: p.label == port, self.ports)
+        interface = port.label
+
+        if channel == 'vlan/native':
+            if new_network is None:
+                return self._remove_native_vlan(interface)
+            else:
+                return self._set_native_vlan(interface, new_network)
+        else:
+            match = re.match(re.compile(r'vlan/(\d+)'), channel)
+            assert match is not None, "HIL passed an invalid channel to the" \
+                " switch!"
+            vlan_id = match.groups()[0]
+
+            if new_network is None:
+                return self._remove_vlan_from_port(interface, vlan_id)
+            else:
+                assert new_network == vlan_id
+                return self._add_vlan_to_trunk(interface, vlan_id)
+
+# 2. Private Methods:
+
+    def _string_to_list(self, a_string):
```

Comment:
As the given in the example (mentioned in the docstring) it parses only these specific kind of strings. So it is not completely generic. Such strings are specific to this driver and may be to Juniper ( but that is another PR) 
Hence I think the function belongs to the class.

Reply:
Here are my 2 cents on this. 

This function doesn't modify/require anything specific to this class. You pass it `a_string` and it does it's thing. So it doesn't make sense to pass it `self`. 

You could make it static, but one could argue that it does nothing specific for this class (for example,. see `validate_port_name` or `construct_tag` in brocade). This is just a general helper function.

Which is why it makes sense to declare it outside the `Ovs()` class *but within this module*, since as you pointed out, it's not very generic. (It also makes for easy testing in my opinion, I dont have to deal with the whole class, just the relevant bits.)

Had it been really generic we would put this function in `hil.ext.switches.common` and import it wherever we need it.

Hope this helps.

Reply:
I see your point and also like the idea of moving it to `hil.ext.switches.common`
since the function is actually needed by the Juniper driver too.
The functions are generic enough for both the drivers to use it.
In case a new use case comes up we can add those functionality in future, thus maintain the 'generality' of the functions. 
In that case I would like to move both of these functions to the `hil.ext.switches.common`. Let me know what you think @naved001